File: bindings/lua/Polycode/QuaternionCurve.lua
class "QuaternionCurve"


function QuaternionCurve:interpolate(t, useShortestPath)
	local retVal = Polycode.QuaternionCurve_interpolate(self.__ptr, t, useShortestPath)
	if retVal == nil then return nil end
	local __c = _G["Quaternion"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function QuaternionCurve:__delete()
	if self then Polycode.delete_QuaternionCurve(self.__ptr) end
end

File: bindings/javascript/Polycode/QuaternionCurve.js
function QuaternionCurve() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.QuaternionCurve()
	}
}


Duktape.fin(QuaternionCurve.prototype, function (x) {
	if (x === QuaternionCurve.prototype) {
		return;
	}
	Polycode.QuaternionCurve__delete(x.__ptr)
})

QuaternionCurve.prototype.interpolate = function(t,useShortestPath) {
	var retVal = new Quaternion("__skip_ptr__")
	retVal.__ptr = Polycode.QuaternionCurve_interpolate(this.__ptr, t, useShortestPath)
	return retVal
}
